src/snowflake/snowpark/query_history.py

```diff
@@ -72,3 +87,32 @@ def include_thread_id(self) -> bool:
     def include_error(self) -> bool:
         """When True, queries that have error during execution are recorded by this listener."""
         return self._include_error
+
+
+class AstListener(QueryListener):
+    def __init__(
+        self,
+        session: "snowflake.snowpark.session.Session",
+        include_failures: bool = False,
```

Comment:
add comment to talk about what `include_failures` does

Reply:
@sfc-gh-oplaton can we add a comment here?